core/src/test/java/hudson/util/AtomicFileWriterTest.java

```diff
@@ -124,4 +140,55 @@ public void badPath() throws Exception {
                        containsString("exists and is neither a directory nor a symlink to a directory"));
         }
     }
+
+    @Issue("JENKINS-48407")
+    @Test
+    public void checkPermissions() throws IOException, InterruptedException {
+
+        final File newFile = tmp.newFile();
+
+        // Check Posix calls are supported (to avoid running this test on Windows for instance)
+        boolean posixSupported = true;
+        try {
+            Files.getPosixFilePermissions(newFile.toPath());
+        } catch (UnsupportedOperationException e) {
+            posixSupported = false;
+        }
+        assumeThat(posixSupported, is(true));
+
+        // given
+        final Set<PosixFilePermission> givenPermissions = EnumSet.of(OWNER_READ,
```

Comment:
The `PosixFilePermissions` class (with an `s`) defines some helper methods that can be useful for this kind of thing: `PosixFilePermissions.fromString("rw-rw-r--")`.

Reply:
Will use next time I need it or if need to refactor this.